presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java

```diff
@@ -2969,4 +2971,17 @@ public FeaturesConfig setInlineProjectionsOnValues(boolean isInlineProjectionsOn
         this.isInlineProjectionsOnValuesEnabled = isInlineProjectionsOnValuesEnabled;
         return this;
     }
+
+    @Config("query-execution-fail-fast-enabled")
```

Comment:
maybe call this `validate-plan-before-queueing`?  Is that what this is for?

Reply:
Yeah, it could be named better. However, it isn't specific for only queries that have been queued. It starts building/validating the plan as soon as `SqlQueryExecution` is constructed instead of waiting until `SqlQueryExecution.start()` is invoked, where `ManagedQueryExecution.startWaitingForResources()` has already completed.

Perhaps something like `eagerly-validate-plan-before-resources` ?

Reply:
IIUC, I think queueing will be a more accurate term, because checking for resources happens after queueing.

Reply:
Sure, my point was that not all queries end up on the queue and might be started right away. In that case this property would still be used and could validate before waiting for resources completes. So I didn't want it to sound like this is only useful if workload has heavy queuing, but if `validate-plan-before-queueing` is the consensus, it's fine with me.

Reply:
Just explaining my thinking a bit here, rather than thinking of queueing as something that happens to a query, I think of it as a component in Presto.  But I see your point as this phrase may not be familiar to folks who are not Presto developers.

What about `eager-plan-validation-enabled`, and in the documentation we describe it more precisely?

Reply:
`eager-plan-validation-enabled` sgtm